src/hooks.c

```diff
@@ -0,0 +1,73 @@
+#include "hooks.h"
+
+unsigned int packet_hook(void *priv, struct sk_buff *skb, const struct nf_hook_state *state) {
+    struct nf_hook_params *params = (struct nf_hook_params *)priv;
+    Packet packet = to_packet(skb);
+    Rule *matched_rule;
+
+    if (packet.ip_header) {
+        matched_rule = get_matching_rule(&packet, params->rules, params->rule_count);
+        if (matched_rule) {
+            print_packet(&packet, KERN_INFO "Packet %pI4:%u to %pI4:%u matched a rule");
+            return handle_packet(&packet, matched_rule);
+        }
+    }
+
+    return NF_ACCEPT;
+}
+
+Rule post_routing_rules[1] = {
+    {
+        {
+            {}, 
+            {ntohl(0xC0A81101), htons(8000)},
+            IPPROTO_TCP
+        },
+        {
+            1, 
+            {
+                {ntohl(0xC0A81180)},
+                {ntohl(0xC0A81101), htons(8080)}
+            }
+        }
+    }
+};
+Rule pre_routing_rules[1] = {
+    {
+        {
+            {ntohl(0xC0A81101), htons(8080)}, 
+            {},
+            IPPROTO_TCP
+        }, 
+        {
+            1, 
```

Comment:
No magic numbers. Define it as a constant (if there is not already one) to indication modify action.
I see you defined it already in rule.h.